src/webgpu/api/operation/sampling/anisotropy.spec.ts

```diff
@@ -6,14 +6,506 @@ Note that anisotropic filtering is never guaranteed to occur, but we might be ab
-  - Result with and without anisotropic filtering is different (if the hardware supports it).
   - Check anisotropy is done in the correct direciton (by having a 2D gradient and checking we get
     more of the color in the correct direction).
-More generally:
-  - Test very large and very small values (even if tests are very weak).
 `;
 
 import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { assert } from '../../../../common/framework/util/util.js';
 import { GPUTest } from '../../../gpu_test.js';
+import { mipSize } from '../../../util/texture/subresource.js';
 
-export const g = makeTestGroup(GPUTest);
+const kRTSize = 16;
+const kBytesPerRow = 256;
+const xm = kRTSize / 2;
+const kColorAttachmentFormat = 'rgba8unorm';
+const kTextureFormat = 'rgba8unorm';
+const colors = [
+  new Uint8Array([0xff, 0x00, 0x00, 0xff]), // miplevel = 0
+  new Uint8Array([0x00, 0xff, 0x00, 0xff]), // miplevel = 1
+  new Uint8Array([0x00, 0x00, 0xff, 0xff]), // miplevel = 2
+];
+const checkerColors = [
+  new Uint8Array([0xff, 0x00, 0x00, 0xff]),
+  new Uint8Array([0x00, 0xff, 0x00, 0xff]),
+];
+
+// renders texture a slanted plane placed in a specific way
+class SamplerAnisotropicFilteringSlantedPlaneTest extends GPUTest {
+  private compareUint8ArrayResult(a: Uint8Array, b: Uint8Array) {
+    if (a.byteLength !== b.byteLength) {
+      return false;
+    }
+    for (let i = 0, len = a.length; i < len; i++) {
+      if (a[i] !== b[i]) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  checkCheckerBoardResults(results: Uint8Array[]) {
+    this.eventualAsyncExpectation(async niceStack => {
+      if (true === this.compareUint8ArrayResult(results[0], results[1])) {
+        niceStack.message =
+          'Render results with sampler.maxAnisotropy being 1 and 16 should be different.';
+        this.rec.expectationFailed(niceStack);
+      }
+      if (false === this.compareUint8ArrayResult(results[1], results[2])) {
+        niceStack.message =
+          'Render results with sampler.maxAnisotropy being 16 and 1024 should be the same.';
+        this.rec.expectationFailed(niceStack);
+      }
+    });
+  }
+
+  copyRenderTargetToBuffer(rt: GPUTexture): GPUBuffer {
+    const byteLength = kRTSize * kBytesPerRow;
+    const buffer = this.device.createBuffer({
+      size: byteLength,
+      usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST,
+    });
+
+    const commandEncoder = this.device.createCommandEncoder();
+    commandEncoder.copyTextureToBuffer(
+      { texture: rt, mipLevel: 0, origin: [0, 0, 0] },
+      { buffer, bytesPerRow: kBytesPerRow, rowsPerImage: kRTSize },
+      { width: kRTSize, height: kRTSize, depth: 1 }
+    );
+    this.queue.submit([commandEncoder.finish()]);
+
+    return buffer;
+  }
+
+  private pipeline: GPURenderPipeline | undefined;
+  private uniformBuffer: GPUBuffer | undefined;
+  private vertexBuffer: GPUBuffer | undefined;
+  setup(): void {
```

Comment:
Should be possible to have this override GPUTest.init and then you won't have to call it in the test function.